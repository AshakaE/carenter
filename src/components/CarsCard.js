import React from 'react';
import PropTypes from 'prop-types';

const CarsCard = (props) => {
  const {
    item: {
      name, model, year, imageUrl,
    },
  } = props;
  return (
    <>
      <div>{name}</div>
      <div>{model}</div>
      <div>{year}</div>
      <div>
        <img src={imageUrl} alt={name} />
      </div>
    </>
  );
};

CarsCard.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string,
    model: PropTypes.string,
    year: PropTypes.number,
    imageUrl: PropTypes.string,
  }),
};

CarsCard.defaultProps = {
  item: [],
};

export default CarsCard;
