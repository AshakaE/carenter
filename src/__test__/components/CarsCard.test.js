import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CarsCard from '../../components/CarsCard';

describe('CarsCard', () => {
  it('renders correctly', () => {
    const id = 1;
    const name = 'test';
    const model = 'test';
    const year = 2021;
    const imageUrl = 'test';
    const tree = renderer
      .create(
        <BrowserRouter>
          <CarsCard
            item={{
              name,
              model,
              imageUrl,
              id,
              year,
            }}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
