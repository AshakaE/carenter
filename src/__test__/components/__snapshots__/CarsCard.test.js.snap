// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsCard renders correctly 1`] = `
<div
  className="cardI"
>
  <div
    className="cardBox"
  >
    <div
      className="cardText"
    >
      <p>
        Name: 
      </p>
      <p>
        test
      </p>
    </div>
    <div
      className="cardText"
    >
      <p>
        model: 
      </p>
      <p>
        test
      </p>
    </div>
    <div
      className="cardText"
    >
      <p>
        year: 
      </p>
      <p>
        2021
      </p>
    </div>
    <div>
      <img
        alt="test"
        src="test"
      />
    </div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      place booking
    </button>
  </div>
</div>
`;
